strax/corrections.py

```diff
@@ -160,25 +183,60 @@ def write(self, correction, df, required_columns=('ONLINE', 'v1')):
         for req in required_columns:
             if req not in df.columns:
                 raise ValueError(f'Must specify {req} in dataframe')
-        # Compare against
+        # Compare against existing data
+        # We can change OFFLINE values in the past only if they are NaN
+        # We cannot change ONLINE values in the past
+        # We can add a new date(row) in the past(ONLINE) as long as it is the same value
         logging.info('Reading old values for comparison')
-        df2 = self.read(correction)
-
-        if df2 is not None:
-            logging.info('Checking if columns unchanged in past')
+        df_old = self.read(correction)
+        if df_old is not None:
             now = datetime.now(tz=timezone.utc)
-            for column in df2.columns:
-                logging.debug(f'Checking {column}')
-                if not (df2.loc[df2.index < now, column] ==
-                        df.loc[df.index < now, column]).all():
-                    raise ValueError(f'{column} changed in past, not allowed')
+            new_dates = df.index.difference(df_old.index)
+            if not new_dates.empty:
+                for i, item in enumerate(new_dates):
+                    if item < now:
+                        for column in df_old.columns:
+                            new_value = df.loc[df.index == new_dates[i].to_pydatetime(), column].values[0]
+                            old_value = df_old.loc[df_old.index < new_dates[i].to_pydatetime(), column][-1]
+                            if new_value != old_value:
```

Comment:
I think here you should also add a check whether new_value and old_value are both NaN, as `np.nan == np.nan` will return False and hence raise a ValueError, which we do not want in that case. Or do I misunderstand the purpose of this check or the possible values of new_value and old_value (at least in the previous version of this check with `(df2.loc[df2.index < now, column] == df.loc[df.index < now, column]).all()` this lead to unwanted errors)?

Reply:
not sure I understand what you are suggesting, I've tested and it works even in the cases of NaNs (as you already know from our slack discussion)